lib/active_fedora/change_set.rb

```diff
@@ -19,7 +19,7 @@ def empty?
 
     def changes
       @changes ||= changed_attributes.each_with_object({}) do |key, result|
-        if m = /^.*_ids?$/.match(key)
+        if (m = /^.*_ids?$/.match(key)) && object.association(m[0].to_sym).present?
```

Comment:
Is the first clause necessary any longer? Did you try changing this line to just have: `object.association(m[0].to_sym).present?`